docs/faq/code-analysis/why-does-codacy-show-unexpected-coverage-changes.md

```diff
@@ -0,0 +1,362 @@
+
+-   **Coverable lines** (lines that can be tested), by listing those lines
+-   **Covered lines** (lines that were tested at least once), by marking those lines as tested or having a number of test hits
+-   **Not coverable lines** (lines that can't be tested), by not listing those lines
+
+For example, the coverage report represented below includes coverage information for two source code files:
+
+-   File `ClassA.java` has two coverable lines and all are covered by tests
+-   File `ClassB.java` has three coverable lines but only line 1 is covered by tests
+
+<table>
+  <thead>
+    <tr>
+      <th>File</th>
+      <th>Line number</th>
+      <th>Covered by tests?</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td rowspan="2">ClassA.java</td>
+      <td>2</td>
+      <td>Yes</td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>Yes</td>
+    </tr>
+    <tr>
+      <td rowspan="3">ClassB.java</td>
+      <td>1</td>
+      <td>Yes</td>
+    </tr>
+    <tr>
+      <td>3</td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>11</td>
+      <td>No</td>
+    </tr>
+  </tbody>
+</table>
+
+Based on the information obtained from the coverage reports, Codacy calculates code coverage as follows:
+
+-   The **coverage** for a file, commit, or pull request is the percentage of covered lines in the universe of coverable lines for that file, commit, or pull request. For example, a commit with 85 covered lines out of a total of 100 coverable lines has 85% coverage.
+
+{%
+    include-markdown "which-metrics-does-codacy-calculate.md"
+    start="<!--start-code-coverage-metrics-->"
+    end="<!--end-code-coverage-metrics-->"
+%}
+
+!!! important
+    Note that changes external to a source code file can affect the lines that are or aren't covered in that file.
+
+## Common reasons for unexpected coverage changes
+
+There are several reasons that could cause Codacy to report unexpected coverage results, from changes to your source code to external factors. The following is a non-exhaustive list of the most common reasons:
+
+-   Adding new tests or removing existing tests from your project.
+
+    Even small and localized changes to your tests can have an impact on the amount of covered lines across your repository.
+
+-   Changing the logic of your application or tests.
+
+    Changing the control flow of your application or tests can mean that different areas of your code start or stop being covered by tests. For example, inverting the result of the Boolean expression of an if statement means that a different branch of your code could now be tested.
+
+-   Failing to upload coverage reports, or uploading a different number of reports between commits.
+
+    This can be caused by a failed step in your CI/CD pipeline, for example. In the case of pull requests, you should make sure that you upload all relevant coverage reports for both the merge base commit and the HEAD commit of the pull request branch.
```

Comment:
Dont know if `merge base commit` is a commonly used name for the `target commit` of the pull request.

Anyways, I had no problem understanding what it meant, just taking the not in case you may find it relevant

Reply:
Interestingly, there doesn't seem to be a "well-agreed" way of calling a commit where a new branch is created. :thinking:

I got the name "merge base commit" from [the Git docs](https://git-scm.com/docs/git-merge-base), but I changed the name to "common ancestor commit" since we were already using that terminology [on this page](https://docs.codacy.com/coverage-reporter/#validating-coverage).